simnode/generator.py

```diff
@@ -85,50 +87,166 @@ def _calculate_parallel_coords(offset: float, line_width: float) \
                     marking_widths = repeat(line_width, num_coords)
                     return list(zip(coords[0], coords[1], z_vals, marking_widths))
 
+                def _calculate_offset_list(relative_offset: float, absolute_offset: float = 0,
+                                           output_number_of_points: int = self.add_roads_to_scenario.num_nodes//3) \
+                                            -> List[float]:
```

Comment:
Since you specify relative_offset and absolute_offset always (except for one time), please remove the default values and insert the default values into the one call which relies on them.
There is no function call which sets output_number_of_points. Additionally it is calculated based on a static variable. So this (basically a constant) value could be another static_variable.